source/scale/high-availability-cluster-based-deployment.rst

```diff
@@ -8,462 +8,414 @@ High availability cluster-based deployment
+.. code-block:: bash
 
-While the connection settings are changing, there might be a brief moment when writes to the master database are unsuccessful. The process waits for all existing connections to finish and starts serving new requests with the new connections. End users attempting to send messages while the switch is happening can have an experience similar to losing connection to the Mattermost server.
+  net.ipv4.ip_local_port_range = 1025 65000
+  net.ipv4.tcp_fin_timeout = 30
+  net.ipv4.tcp_tw_reuse = 1
+  net.core.somaxconn = 4096
+  net.ipv4.tcp_max_syn_backlog = 8192
+  vm.min_free_kbytes = 167772
+  net.ipv4.tcp_congestion_control = bbr
+  net.core.default_qdisc = fq
+  net.ipv4.tcp_mem = 1638400 1638400 1638400
 
-Transparent failover
-````````````````````
+Apply similar optimizations to your proxy server.
 
-The database can be configured for high availability and transparent failover use the existing database technologies. We recommend PostgreSQL Clustering or Amazon Aurora. Database transparent failover is beyond the scope of this documentation.
+Cluster discovery
+~~~~~~~~~~~~~~~~~
 
-Recommended configuration settings for PostgreSQL
-``````````````````````````````````````````````````
+After the basic server configuration, enable node discovery to ensure servers in the cluster can find and communicate with each other.
 
-If you're using PostgreSQL as the choice of database, we recommend the following configuration optimizations on your Mattermost server. These configurations were tested on an AWS Aurora r5.xlarge instance of PostgreSQL 11.7. There are also some general optimizations mentioned which requires servers with higher specifications.
+If using custom network configurations:
 
-**Config for Postgres Primary or Writer node**
+- Use :ref:`OverrideHostname <configure/environment-configuration-settings:override hostname>` for per-server configurations.
+- Ensure :ref:`UseIpAddress <configure/environment-configuration-settings:use ip address>` is set to ``true`` for automatic discovery of the first non-local IP.
 
-.. code-block:: sh
+Verify discovery settings via the database: ``SELECT * FROM ClusterDiscovery;``
 
-  # If the instance is lower capacity than r5.xlarge, then set it to a lower number. 
-  # Also tune the "MaxOpenConns" setting under the "SqlSettings" of the Mattermost app accordingly. 
-  # Note that "MaxOpenConns" on Mattermost is per data source name.
-  max_connections = 1024
+Ensure necessary ports are open for inter-node communication.
 
-  # Set it to 1.1, unless the DB is using spinning disks.
-  random_page_cost = 1.1
+State synchronization
+~~~~~~~~~~~~~~~~~~~~~
 
-  # This should be 32MB if using read replicas, or 16MB if using a single PostgreSQL instance. 
-  # If the instance is of a lower capacity than r5.xlarge, then set it to a lower number.
-  work_mem = 32MB
+Once nodes can communicate, synchronize session data and WebSocket messages across nodes through inter-node communication.
```

Comment:
This is not something that admins need to do, it's how Mattermost works. The `State` section in the old document is purely informative in that regard, but I still think it's useful.